gcc/rust/ast/rust-ast.h

```diff
@@ -32,6 +32,10 @@ typedef std::string Identifier;
 typedef int TupleIndex;
 struct Session;
 struct MacroExpander;
+// forward declaration Dump to pass the compilation
+namespace HIR {
+class Dump;
```

Comment:
This "contamination" is not great, but it is the best we can do right now.

Reply:
Refactor is neccessary in the future.